### PR TITLE
Fix shuffle task on oscar

### LINKED ISSUE #2145: [BUG] Shuffle task raised `KeyError`
**Describe the bug**
Shuffle task raised `KeyError` when cluster has more than one workers.

``` Python
s = np.random.RandomState(0)
raw = pd.DataFrame(s.rand(100, 4), columns=list('abcd'))
df = md.DataFrame(raw, chunk_size=20)
df.sort_values(by='a').execute()
```

error message:

```
../session.py:151: in _run_in_background
    raise task_result.error.with_traceback(task_result.traceback)
../../../services/scheduling/worker/execution.py:223: in internal_run_subtask
    yield asyncio.shield(run_aiotask)
mars/oscar/core.pyx:252: in mars.oscar.core._Actor._run_actor_async_generator
    ???
mars/oscar/core.pyx:194: in _handle_actor_result
    ???
../../../services/subtask/api.py:52: in run_subtask_in_slot
    return await ref.run_subtask(subtask)
../../../oscar/backends/context.py:149: in send
    return self._process_result_message(result)
../../../oscar/backends/context.py:58: in _process_result_message
    raise message.error.with_traceback(message.traceback)
../../../oscar/backends/pool.py:481: in send
    result = await future
../../../oscar/api.py:111: in __on_receive__
    return await super().__on_receive__(message)
mars/oscar/core.pyx:302: in __on_receive__
    ???
mars/oscar/core.pyx:194: in _handle_actor_result
    ???
../../../services/subtask/worker/subtask.py:440: in _run_subtask
    return await processor.run()
../../../services/subtask/worker/subtask.py:227: in run
    await future
/Users/hekaisheng/miniconda3/envs/py3.8/lib/python3.8/concurrent/futures/thread.py:57: in run
    result = self.fn(*self.args, **self.kwargs)
../../../services/subtask/worker/subtask.py:194: in _execute_operand
    return execute(ctx, op)
../../../core/operand/core.py:347: in execute
    return executor(results, op)
../../../dataframe/sort/psrs.py:645: in execute
    cls._execute_reduce(ctx, op)
../../../dataframe/sort/psrs.py:610: in _execute_reduce
    raw_inputs = list(op.iter_mapper_data(ctx, pop=False))
../../../core/operand/shuffle.py:90: in iter_mapper_data
    for _idx, data in self.iter_mapper_data_with_index(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   yield idx, ctx[key, self.reducer_index]
E   KeyError: ('d357be89b8d14061e80d86e321d793a1', (2,))

../../../core/operand/shuffle.py:81: KeyError
```

### PR DESCRIPTION
**Pull request by hekaisheng**
## What do these changes do?

For shuffle task, we decref mapper chunks when shuffle proxy finishes, it leads to `KeyError` as reducer chunks depends on mapper chunks. This PR fixes by incref mapper chunks when processing reducers.

## Related issue number

Fixes #2145

### CODE REVIEW

File: mars/services/storage/core.py [hunk 1]
```diff
@@ -138,6 +138,9 @@ def __init__(self):
         # mapping value is list of InternalDataInfo
         self._data_key_to_info: Dict[tuple, List[InternalDataInfo]] = defaultdict(list)
         self._data_info_list = dict()
+        # data key may be a tuple in some cases,
+        # we record main key to manage their lifecycle
+        self._tuple_keys = defaultdict(set)
```
Comment (qinxuye) [inline mars/services/storage/core.py:143]: Rename it with a more reasonable name, maybe `_main_key_to_sub_keys`?

**Review by qinxuye — COMMENTED** [review]
LGTM overall.

**Review by qinxuye — APPROVED** [review]
LGTM

File: mars/services/storage/core.py [hunk 2]
```diff
@@ -575,10 +604,14 @@ async def fetch(self,
                         main_key = data_key[0] if isinstance(data_key, tuple) else data_key
                         address = (await meta_api.get_chunk_meta(
                             main_key, fields=['bands']))['bands'][0][0]
+                    if address == self.address:
+                        return
+                    logger.info(f'Begin to fetch data {data_key} from {address}')
```
Comment (wjsi) [inline mars/services/storage/core.py:609]: Switch log level into DEBUG

File: mars/deploy/oscar/tests/test_local.py [hunk 3]
```diff
@@ -92,6 +92,23 @@ async def test_iterative_tiling(create_cluster):
     assert df2.index_value.max_val <= 30
 
 
+@pytest.mark.asyncio
+async def test_execute_shuffle(create_cluster):
```
Comment (wjsi) [inline mars/deploy/oscar/tests/test_local.py:96]: Use ``test_execute_describe``.

**Review by wjsi — CHANGES_REQUESTED** [review]

**Review by wjsi — APPROVED** [review]
LGTM

**Review by qinxuye — APPROVED** [review]
LGTM

**Pull request merged**